app/upload/upload.component.ts

```diff
@@ -0,0 +1,20 @@
+import {Component} from 'angular2/core';
+import {UploadFileSelect} from './upload-file-select.directive';
+import {UploadService} from './upload.service';
+import {MimeTypeService} from '../../util/mime-type.service';
+import {HTTP_PROVIDERS} from 'angular2/http';
+import {FileUpload} from './file-upload.component';
+
+@Component({
+  template: `
+    <input type="file" upload-file />
+    <file-upload *ngFor="#upload of uploadService.uploads" [upload]="upload">
+    </file-upload>
+  `,
+  providers: [HTTP_PROVIDERS, UploadService, MimeTypeService],
```

Comment:
MimeTypeService is needed here so it can be listed in the providers so sub components can have it injected. HTTP_PROVIDERS is left over from when I was implementing a version of this using the Http service - I'll remove it.